plugins/optimization-detective/detect.js

```diff
@@ -308,14 +344,55 @@ export default async function detect( {
 		);
 	}
 
+	if ( win.innerWidth === 0 || win.innerHeight === 0 ) {
+		if ( isDebug ) {
+			log(
+				'Window must have non-zero dimensions for URL Metric collection.'
+			);
+		}
+		return;
+	}
+
 	// Abort if the current viewport is not among those which need URL Metrics.
-	if ( ! isViewportNeeded( win.innerWidth, urlMetricGroupStatuses ) ) {
+	const urlMetricGroupStatus = getGroupForViewportWidth(
+		win.innerWidth,
+		urlMetricGroupStatuses
+	);
+	if ( urlMetricGroupStatus.complete ) {
 		if ( isDebug ) {
 			log( 'No need for URL Metrics from the current viewport.' );
 		}
 		return;
 	}
 
+	// Abort if the client already submitted a URL Metric for this URL and viewport group.
+	// TODO: Remove the console timing.
+	console.time( 'sha1' ); // eslint-disable-line no-console
+	const alreadySubmittedSessionStorageKey =
+		await getAlreadySubmittedSessionStorageKey(
+			currentETag,
+			currentUrl,
+			urlMetricGroupStatus
+		);
+	console.timeEnd( 'sha1' ); // eslint-disable-line no-console
```

Comment:
@felixarntz What are your thoughts on the hashing of the key? The timing shows that this takes 10ms. Nevertheless, it is not blocking the main thread from. With 20x CPU slowdown it takes 35 ms:

![image](https://github.com/user-attachments/assets/842cab50-f0b8-4e8e-bf2f-b3d8476921d2)

See this trace: https://trace.cafe/t/CNJQYk4RTE

I also tried computing the hash 100 times in this function to better confirm that it is not blocking the main thread: https://trace.cafe/t/mBlNPnrYft

![image](https://github.com/user-attachments/assets/cf2a4bce-a515-4fa0-88d6-c41c73e51a7c)

So if this seems good to you and that hashing is warranted, then this suggestion can be accepted:

```suggestion
	const alreadySubmittedSessionStorageKey =
		await getAlreadySubmittedSessionStorageKey(
			currentETag,
			currentUrl,
			urlMetricGroupStatus
		);
```

Reply:
Thanks for the detailed research, that works for me! 👍